pkg/rosa/client_test.go

```diff
@@ -0,0 +1,139 @@
+			"ocmApiUrl":       []byte(url),
+			"ocmClientId":     []byte(clientId),
+			"ocmClientSecret": []byte(clientSecret),
+		},
+	}
+}
+
+func createCP(namespace string) *rosacontrolplanev1.ROSAControlPlane {
+	return &rosacontrolplanev1.ROSAControlPlane{
+		Spec: rosacontrolplanev1.RosaControlPlaneSpec{
+			CredentialsSecretRef: &corev1.LocalObjectReference{
+				Name: "rosa-creds-secret",
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+		},
+	}
+
+}
+
+func TestOcmCredentials(t *testing.T) {
+	g := NewWithT(t)
+
+	wlSecret := createSecret("rosa-creds-secret", "default", "", "url", "client-id", "client-secret")
+	mgrSecret := createSecret("rosa-creds-secret", system.GetManagerNamespace(), "", "url", "global-client-id", "global-client-secret")
+
+	cp := createCP("default")
+
+	// Test that ocmCredentials() prefers workload secret to global and environment secrets
+	cp.Spec = rosacontrolplanev1.RosaControlPlaneSpec{
+		CredentialsSecretRef: &corev1.LocalObjectReference{
+			Name: "rosa-creds-secret",
+		},
+	}
+	os.Setenv("OCM_API_URL", "env-url")
+	os.Setenv("OCM_TOKEN", "env-token")
+	rcpScope := createROSAControlPlaneScope(wlSecret, mgrSecret, cp)
+	token, url, clientId, clientSecret, err := ocmCredentials(context.Background(), rcpScope)
+
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(token).To(Equal(string(wlSecret.Data["ocmToken"])))
+	g.Expect(url).To(Equal(string(wlSecret.Data["ocmApiUrl"])))
+	g.Expect(clientId).To(Equal(string(wlSecret.Data["ocmClientId"])))
+	g.Expect(clientSecret).To(Equal(string(wlSecret.Data["ocmClientSecret"])))
+
+	// Test that ocmCredentials() prefers global manager secret to environment secret in case workload secret is not specified
+	cp.Spec = rosacontrolplanev1.RosaControlPlaneSpec{}
+	rcpScope = createROSAControlPlaneScope(wlSecret, mgrSecret, cp)
+	token, url, clientId, clientSecret, err = ocmCredentials(context.Background(), rcpScope)
+
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(token).To(Equal(string(mgrSecret.Data["ocmToken"])))
+	g.Expect(url).To(Equal(string(mgrSecret.Data["ocmApiUrl"])))
+	g.Expect(clientId).To(Equal(string(mgrSecret.Data["ocmClientId"])))
+	g.Expect(clientSecret).To(Equal(string(mgrSecret.Data["ocmClientSecret"])))
+
+	// Test that ocmCredentials() returns environment secret in case workload and manager secret are not specified
+	cp.Spec = rosacontrolplanev1.RosaControlPlaneSpec{}
+	rcpScope = createROSAControlPlaneScope(nil, nil, cp)
+	token, url, clientId, clientSecret, err = ocmCredentials(context.Background(), rcpScope)
+
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(token).To(Equal(string(os.Getenv("OCM_TOKEN"))))
+	g.Expect(url).To(Equal(string(os.Getenv("OCM_API_URL"))))
+	g.Expect(clientSecret).To(Equal(""))
+
+	// Test that ocmCredentials() returns error in case none of the secrets has been provided
+	cp.Spec = rosacontrolplanev1.RosaControlPlaneSpec{}
+	rcpScope = createROSAControlPlaneScope(nil, nil, cp)
```

Comment:
why do you create a dummy secrets in `createROSAControlPlaneScope` that you don't need in the test

Reply:
Yea, I rewrote `createROSAControlPlaneScope()` as varadic to make this more simple.